src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -468,6 +468,7 @@ internal override void ForceComplete(SourceLocation locationOpt, CancellationTok
                             var diagnostics = DiagnosticBag.GetInstance();
                             AfterMembersChecks(diagnostics);
                             AddDeclarationDiagnostics(diagnostics);
+                            DeclaringCompilation.SymbolDeclaredEvent(this);
```

Comment:
Sure, I will document how the placement of this call relaxes the ordering constraints.